src/java_bytecode/library/Makefile

```diff
@@ -1,26 +1,46 @@
-all: org.cprover.jar
+include ../../config.inc
+include ../../common
 
 SOURCE_DIR := src
 BINARY_DIR := classes
 
 FIND := find
 
 JAVAC := javac
-JFLAGS := -sourcepath $(SOURCE_DIR) -d $(BINARY_DIR)
+JFLAGS := -sourcepath $(SOURCE_DIR) -d $(BINARY_DIR) -XDignore.symbol.file
 
 CLASSPATH := SOURCE_DIR
 
-all_javas := $(shell $(FIND) $(SOURCE_DIR) -name '*.java')
+ALL_JAVAS := $(wildcard $(SOURCE_DIR)/*/*.java $(SOURCE_DIR)/*/*/*.java $(SOURCE_DIR)/*/*/*/*.java) #$(shell find $(SOURCE_DIR) -name '*.java')
+ALL_CLASSES := $(patsubst $(SOURCE_DIR)/%.java,$(BINARY_DIR)/%.class,$(ALL_JAVAS))
 
 $(BINARY_DIR):
 	mkdir -p $(BINARY_DIR)
 
-.PHONY: compile
-compile: $(BINARY_DIR)
-	$(JAVAC) $(JFLAGS) $(all_javas)
+.SUFFIXES: .java .class
+
+$(BINARY_DIR)/%.class : $(SOURCE_DIR)/%.java $(BINARY_DIR)
+	$(JAVAC) $(JFLAGS) $(patsubst $(BINARY_DIR)/%.class,$(SOURCE_DIR)/%.java,$@)
 
 JAR := jar
 JARFLAGS := -cf
 
-org.cprover.jar: compile
-	$(JAR) $(JARFLAGS) $@ -C $(BINARY_DIR) org
+.NOTPARALLEL: core-models.jar
+core-models.jar: $(ALL_CLASSES)
+	$(JAR) $(JARFLAGS) $@ -C $(BINARY_DIR) .
+
+CLEANFILES = core-models.jar converter java_core_models.inc
+
+clean : clean_
+
+clean_ :
+	$(RM) -Rf $(BINARY_DIR)
+
+all: java_core_models.inc
+
+converter: converter.cpp
```

Comment:
Write `SRC = converter.cpp` instead of giving a explicit rule to build the converter, as it is done elsewhere in the codebase to compile .cpp files to .o files. Move this line to the top of the file, as in other Makefiles. That way 'make clean' will also remove converter.d, which wasn't being removed.

Reply:
done